mesonbuild/build.py

```diff
@@ -619,8 +619,11 @@ def construct_id_from_path(subdir: str, name: str, type_suffix: str) -> str:
         return my_id
 
     def get_id(self) -> str:
+        name = self.name
+        if hasattr(self, 'name_suffix_set') and self.name_suffix_set:
+            name = name + '.' + self.suffix
```

Comment:
I think you can use `self.filename` instead, it already contains prefix and suffix.

Reply:
And in that case, do we even need to include type_suffix() at all? I think (subdir, filename) pair must always be unique, otherwise ninja targets would class anyway.

Reply:
Unfortunately, the `post_init` for the class isn't run at this stage yet so `self.filename` is `no_name` here.